.github/workflows/test.yml

```diff
@@ -15,7 +15,7 @@ jobs:
       - name: 'Checkout Repository'
         uses: actions/checkout@v3
       - name: 'Dependency Review'
-        uses: actions/dependency-review-action@v3
+        uses: actions/dependency-review-action@v4
```

Comment:
Old and was failing, hence the upgrade.

Reply:
Alright this one's still [failing](https://github.com/novuhq/novu/actions/runs/9136987610/job/25126425904?pr=5598#step:3:5). Not sure why this has broken, needs more investigation.

Reply:
Tried running this step ([dependency-review-action](https://github.com/actions/dependency-review-action)) with debug logging, here's the [logs](https://github.com/novuhq/novu/actions/runs/9150672174/job/25155779581?pr=5598). Didn't give much clues to the failure reason.

@scopsy is this a familiar error during PNPM upgrades?

_Logs_
```bash
##[debug]Evaluating condition for step: 'Dependency Review'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Dependency Review
##[debug]Loading inputs
##[debug]Evaluating: github.token
##[debug]Evaluating Index:
##[debug]..Evaluating github:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'token'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Loading env
Run actions/dependency-review-action@v[4](https://github.com/novuhq/novu/actions/runs/9150672174/job/25155779581?pr=5598#step:3:4)
  with:
    repo-token: ***
  
Error: Server Error
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Dependency Review
```

Reply:
Similar issue [here](https://github.com/actions/dependency-review-action/issues/392#issuecomment-1420957190) with the generic `Server Error` error. The recommendation is to try the Dependency Review API directly to verify if the error occurs without the action. Trying that now. If the error is present still, I suspect the Dependency Review API is not compatible with the latest PNPM lockfile `v9`.

Reply:
The error is coming from the Dependency Review API:

```bash
> curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer 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" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/novuhq/novu/dependency-graph/compare/next...64426d4ce6497c0331867df156775714808a8acf
{
  "message": "Server Error"
}
```

Investigating to see if the error resolves after comparing equivalent lockfile versions.

Reply:
The error is coming from the Dependency Review API:

```bash
> curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer XXX" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/novuhq/novu/dependency-graph/compare/next...64426d4ce6497c0331867df156775714808a8acf
{
  "message": "Server Error"
}
```

Investigating to see if the error resolves after comparing equivalent lockfile versions.

Reply:
Using the last 2 commits of this PR, I have verified that the issue appears to occur only when using commits with different lockfile versions. A comparison of the last 2 commits yields a successful API request.

```bash
> curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer XXX" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/novuhq/novu/dependency-graph/compare/0440e0ed331e66e6aa8a7502ba241d1921d71bdd...64426d4ce6497c0331867df156775714808a8acf
[

]
```

We can safely ignore this error and a steady state will be reached in subsequent PRs.